controllers/articles.controllers.js

```diff
@@ -38,13 +43,19 @@ exports.getCommentsByArticleId = (req, res, next) => {
 };
 
 exports.postCommentToArticle = (req, res, next) => {
-    const {article_id} = req.params;
-    const {body} = req
-    addCommentToArticle(article_id, body)
-    .then((comment) => {
-        res.status(201).send({comment})
+  const { article_id } = req.params;
+  const { body } = req;
+  const { username } = body
+  const checkArticle = checkArticleExists(article_id)
+  const checkUser = checkUserExists(username)
+  const checkBody = checkCommentBodyFormat(body)
```

Comment:
SQL would also handle this - if the body is the wrong data type it will error at the query stage

Reply:
I tried this one without the check and SQL added the object into the database as a string. I'm unsure if this would be a problem, but I guess if someone was interacting with the API programmatically you would want them to use only string values and not objects.